nix/modules/fedimintd.nix

```diff
@@ -0,0 +1,233 @@
+            '';
+          }
+          {
+            assertion = cfg.api.address != null;
+            message = ''
+              `services.fedimintd.${fedimintdName}.api.address` must be set to address reachable by the clients, with TLS terminated by external service (typically nginx), and relayed to the fedimintd bind address.
+
+              Example: `wss://api.mymint.org`.
+            '';
+          }
+        ])
+        eachFedimintd);
+
+    networking.firewall.allowedTCPPorts = flatten
+      (mapAttrsToList
+        (fedimintdName: cfg:
+          (
+            if cfg.api.openFirewall then [
+              cfg.api.port
+            ] else [ ]
+          ) ++ (
+            if cfg.p2p.openFirewall then [
+              cfg.p2p.port
+            ] else [ ]
+          )
+        )
+        eachFedimintd);
+
+
+    systemd.services = mapAttrs'
+      (fedimintdName: cfg: (
+        nameValuePair "fedimintd-${fedimintdName}" (
+
+          {
+            description = "Fedimint Server";
+            documentation = [ "https://github.com/fedimint/fedimint/" ];
+            wantedBy = [ "multi-user.target" ];
+            environment = lib.mkMerge ([
+              {
+                RUST_LOG = cfg.rustLogEnv;
+                FM_BIND_P2P = "${cfg.p2p.bind}:${builtins.toString cfg.p2p.port}";
+                FM_BIND_API = "${cfg.api.bind}:${builtins.toString cfg.api.port}";
+                FM_P2P_URL = cfg.p2p.address;
+                FM_API_URL = cfg.api.address;
+                FM_DATA_DIR = cfg.dataDir;
+                FM_BITCOIN_NETWORK = cfg.bitcoin.network;
+                FM_BITCOIN_RPC_URL = cfg.bitcoin.rpc.address;
+                FM_BITCOIN_RPC_KIND = cfg.bitcoin.rpc.kind;
+              }
+              cfg.extraEnvironment
+            ]);
+            serviceConfig = {
+              DynamicUser = true;
+              User = "fedimint";
+              LockPersonality = true;
+              MemoryDenyWriteExecute = true;
+              ProtectClock = true;
+              ProtectControlGroups = true;
+              ProtectHostname = true;
+              ProtectKernelLogs = true;
+              ProtectKernelModules = true;
+              ProtectKernelTunables = true;
+              PrivateDevices = true;
+              PrivateMounts = true;
+              PrivateUsers = true;
+              RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
+              RestrictNamespaces = true;
+              RestrictRealtime = true;
+              SystemCallArchitectures = "native";
+              SystemCallFilter = [
+                "@system-service"
+                "~@privileged"
+              ];
+              StateDirectory = "fedimintd";
+              StateDirectoryMode = "0700";
+              ExecStart = "${cfg.package}/bin/fedimintd";
+              Restart = "always";
+              RestartSec = 10;
+              StartLimitBurst = 5;
+              UMask = "077";
+              LimitNOFILE = "100000";
+            };
```

Comment:
did you try running this, I wouldn't be surprised something got broken by systemd sandbox